### DISCUSSION TITLE
Python library for easy multispeaker TTS synthesizing and HTML/EPUB conversion

### DISCUSSION THREAD

**Discussion by knochenhans** [coqui-ai/TTS#2580] [Show and tell]
Hi, I’m working on a Python library that aims to make it as easy as possible to synthesize complex multispeaker text with Coqui TTS and thought this might be helpful to someone here. My main goal is to create an easy to use yet customizable framework for writing audiobook files from different document formats like HTML, EPUB, or just about anything that might be needed.

You can check out the current version and more info here: [https://github.com/knochenhans/tts_arranger](https://github.com/knochenhans/tts_arranger)

Here is what a simple example would look like (writes an m4b audiobook containing 2 chapters, metadata and cover image):
```python
# English example using tts_models/en/vctk/vits (with multispeaker support)

tts_items = []

preferred_speakers = ['p273', 'p330']

items1 = []
items1.append(TTS_Item('This is a test', 0))  # Uses preferred speaker #0
items1.append(TTS_Item(length=2000))  # Insert pause
items1.append(TTS_Item('This is a test with another speaker and a fixed minimum length', 1, length=10000))  # Uses preferred speaker #1

items2 = []
items2.append(TTS_Item('Another test',  0))
items2.append(TTS_Item('This is getting boring!', 1))

chapter = []
chapter.append(TTS_Chapter(items1, 'Chapter 1'))
chapter.append(TTS_Chapter(items2, 'Chapter 2'))

project = TTS_Project(chapter, 'Project Title', 'Project Subtitle', author='Some Author')

# Add a cover image
project.add_image_from_url('https://coqui.ai/static/38a06ec53309f617be3eb3b8b9367abf/598c3/logo-wordmark.png')

writer = TTS_Writer(project, '/tmp/tts_arranger_example_output/'), preferred_speakers=preferred_speakers)
writer.synthesize_and_write(project.author + ' - ' + project.title)
```

HTML and EPUB conversion is also possible, but I haven’t come around to document the workflow properly. I’m also working on a Qt based GUI that would allow easy file conversion into audiobooks via drag & drop, and a system to automatically retrieve blogs and convert them into audiobooks, but this it not yet ready to be released.

I’m happy for any kind of feedback. Oh and this is my first Python project of this scope, so you’ll probably run into bugs 😅

**Comment by erogol** [comment]
This is cool! Thanks for sharing.

**Comment by konashevich** [comment]
@knochenhans how's your project going? I saw that you published the library, but did you succeed with the UI? Do you have any plans to continue the project?
